codeserver/ubi9-python-3.9/run-code-server.sh

```diff
@@ -16,6 +16,12 @@ fi
 # Initilize access logs for culling
 echo '[{"id":"code-server","name":"code-server","last_activity":"'$(date -Iseconds)'","execution_state":"running","connections":1}]' > /var/log/nginx/vscode.access.log
 
+# Install extensions if not present
+if [ ! -f "/opt/app-root/src/.local/share/code-server" ]; then
+   code-server --install-extension ms-python.python
+   code-server --install-extension ms-toolsai.jupyter
+fi
```

Comment:
Instead of doing this on run time, we should probably do this in build time.
as this would fail in disconnected environment.

```
RUN code-server --install-extension ms-python.python && \
   code-server --install-extension ms-toolsai.jupyter
```
in the dockerfile.
https://github.com/opendatahub-io/notebooks/blob/271208a7754c5ed29e02aeaac981fc80d68c08de/codeserver/ubi9-python-3.9/Dockerfile#L89

thoughts?

Reply:
I tried to install them on build run, but when you lunch the image on ODH, the extensions are not there, probably because the user's PVC is mounted at /opt/app-root/src, and it's "masked" and replaced by the volume that was mounted. Maybe, I should investigate the case to download the extension's files.

Reply:
Used .vsix packages for ms-python and ms-toolsai to be compatible with disconnected environments. I guess we will be safe now!

Reply:
Thanks for the work.
:100: